exhibitor-core/src/main/java/com/netflix/exhibitor/core/config/azure/AzureConfigProvider.java

```diff
@@ -0,0 +1,147 @@
+package com.netflix.exhibitor.core.config.azure;
+
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.BlobProperties;
+import com.microsoft.azure.storage.blob.CloudBlob;
+import com.netflix.exhibitor.core.azure.*;
+import com.netflix.exhibitor.core.config.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Properties;
+
+public class AzureConfigProvider implements ConfigProvider {
+    private final AzureConfigArguments arguments;
+    private final AzureClient azureClient;
+    private final String hostname;
+    private final Properties defaults;
+
+    /**
+     * @param factory     the factory
+     * @param credentials credentials
+     * @param arguments   args
+     * @param hostname    this VM's hostname
+     * @throws Exception errors
+     */
+    public AzureConfigProvider(AzureClientFactory factory, AzureCredential credentials, AzureConfigArguments arguments, String hostname) throws Exception {
+        this(factory, credentials, arguments, hostname, new Properties());
+    }
+
+    /**
```

Comment:
this is a fairly complex set of constructor arguments, it may be worth, in the javadoc, to provide an example of how to get/use all this stuff; in particular, I'd be looking here for info as to use this class The Right Way